src/main/cpp/DragonVision/DragonVision.cpp

```diff
@@ -277,121 +277,160 @@ std::optional<VisionData> DragonVision::GetVisionDataFromNote(VISION_ELEMENT ele
+		launcherMultiTag = dynamic_cast<DragonPhotonCam *>(m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::LAUNCHER])->GetMultiTagEstimate();
+	}
 
-			// calculate transform to fieldElement as difference between robot pose and field element pose
-			frc::Transform3d transformToElement = frc::Transform3d{robotPose, fieldElementPose};
+	if (m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::PLACER] != nullptr)
+	{
+		placerMultiTag = dynamic_cast<DragonPhotonCam *>(m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::LAUNCHER])->GetMultiTagEstimate();
+	}*/
 
-			// calculate rotation3d for angles from robot center, not transformation
-			units::angle::radian_t pitch = units::math::atan2(transformToElement.Z(), transformToElement.X());
-			units::angle::radian_t yaw = units::math::atan2(transformToElement.Y(), transformToElement.X());
-			units::angle::radian_t roll = units::math::atan2(transformToElement.Z(), transformToElement.Y());
-			frc::Rotation3d rotation = frc::Rotation3d(roll, pitch, yaw);
+	std::optional<VisionPose> launcherMultiTag = dynamic_cast<DragonPhotonCam *>(m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::LAUNCHER])->GetMultiTagEstimate();
+	std::optional<VisionPose> placerMultiTag = dynamic_cast<DragonPhotonCam *>(m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::PLACER])->GetMultiTagEstimate();
+	std::optional<VisionPose> selectedPose = std::nullopt;
 
-			// rebundle into vision data with april tag thats used
-			return VisionData{transformToElement, transformToElement.Translation(), rotation, -1};
-		}
+	if (launcherMultiTag && placerMultiTag)
+	{
+		double launcherAmbiguity = launcherMultiTag.value().visionMeasurementStdDevs[0];
+		double placerAmbiguity = placerMultiTag.value().visionMeasurementStdDevs[0];
+
+		selectedPose = launcherAmbiguity <= placerAmbiguity ? launcherMultiTag.value() : placerMultiTag.value(); // if launcher is less ambiguous, select it, and vice versa
+	}
+	else if (!launcherMultiTag && !placerMultiTag)
+	{
+		return std::nullopt;
+	}
+	else
+	{
+		if (launcherMultiTag)
+			selectedPose = launcherMultiTag.value();
+
+		else if (placerMultiTag)
+			selectedPose = placerMultiTag.value();
+	}
+
+	if (selectedPose)
+	{
+		// calculate transform to fieldElement as difference between robot pose and field element pose
+		frc::Transform3d transformToElement = frc::Transform3d{selectedPose.value().estimatedPose, elementPose};
+
+		// calculate rotation3d for angles from robot center, not transformation
+		units::angle::radian_t pitch = units::math::atan2(transformToElement.Z(), transformToElement.X());
+		units::angle::radian_t yaw = units::math::atan2(transformToElement.Y(), transformToElement.X());
+		units::angle::radian_t roll = units::math::atan2(transformToElement.Z(), transformToElement.Y());
+		frc::Rotation3d rotation = frc::Rotation3d(roll, pitch, yaw);
+
+		// rebundle into vision data with april tag thats used
+		return VisionData{transformToElement, transformToElement.Translation(), rotation, -1};
```

Comment:
should april tag id be -1 here?

Reply:
Currently getting the multitag estimate from a PhotonCam does not give out an apriltag id.  This can be updated later to return the first apriltag id it used, or a vector of all ids used to estimate the pose.